packages/core/src/domain/error/error.ts

```diff
@@ -65,6 +67,10 @@ function hasUsableStack(isErrorInstance: boolean, stackTrace?: StackTrace): stac
   return stackTrace.stack.length > 0 && (stackTrace.stack.length > 1 || stackTrace.stack[0].url !== undefined)
 }
 
+export function isErrorWithFingerprint(error: unknown): error is { dd_fingerprint: unknown } {
+  return error instanceof Error && 'dd_fingerprint' in error
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this function is always used as `fingerprint = isErrorWithFingerprint(error) ? String(error.dd_fingerprint) : undefined`. Maybe replace this function with `getErrorFingerprint(error): string | undefined` returning the sanitized fingertip directly?

Reply:
Good idea!